From 9322566884bdd948c6231ce8e660fc2665698b32 Mon Sep 17 00:00:00 2001
From: Zubair Lutfullah Kakakhel <zubair@resin.io>
Date: Wed, 19 Sep 2018 10:53:04 +0100
Subject: [PATCH] bootcount: Add support to write bootcount to any filesystem

The driver limits bootcount writes to EXT filesystem.
Modify the flags so that the bootcount file can be written to any
filesystem.

Tested on a fat filesystem on a raspberry pi 3

Signed-off-by: Zubair Lutfullah Kakakhel <zubair@resin.io>
Upstream-Status: Inappropriate [pending]
---
 drivers/bootcount/bootcount_ext.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/bootcount/bootcount_ext.c b/drivers/bootcount/bootcount_ext.c
index 075e590..ff25a10 100644
--- a/drivers/bootcount/bootcount_ext.c
+++ b/drivers/bootcount/bootcount_ext.c
@@ -16,7 +16,7 @@ void bootcount_store(ulong a)
 	int ret;
 
 	if (fs_set_blk_dev(CONFIG_SYS_BOOTCOUNT_EXT_INTERFACE,
-			   CONFIG_SYS_BOOTCOUNT_EXT_DEVPART, FS_TYPE_EXT)) {
+			   CONFIG_SYS_BOOTCOUNT_EXT_DEVPART, FS_TYPE_ANY)) {
 		puts("Error selecting device\n");
 		return;
 	}
@@ -39,7 +39,7 @@ ulong bootcount_load(void)
 	int ret;
 
 	if (fs_set_blk_dev(CONFIG_SYS_BOOTCOUNT_EXT_INTERFACE,
-			   CONFIG_SYS_BOOTCOUNT_EXT_DEVPART, FS_TYPE_EXT)) {
+			   CONFIG_SYS_BOOTCOUNT_EXT_DEVPART, FS_TYPE_ANY)) {
 		puts("Error selecting device\n");
 		return 0;
 	}
-- 
2.7.4

